source/simulator/helper_functions.cc

```diff
@@ -2117,6 +2118,57 @@ namespace aspect
   }
 
 
+
+  template <int dim>
+  void
+  Simulator<dim>::fallback_to_amg_preconditioner_if_necessary()
+  {
+    if (parameters.include_melt_transport == true)
+      {
+        parameters.stokes_solver_type = Parameters<dim>::StokesSolverType::block_amg;
+
+        pcout << "Warning, the GMG preconditioner is not supported for models with two-phase flow. Disabling GMG preconditioner." << std::endl;
+      }
+    else if (parameters.material_averaging == MaterialModel::MaterialAveraging::none)
+      {
+        parameters.stokes_solver_type = Parameters<dim>::StokesSolverType::block_amg;
+
+        pcout << "Warning, the GMG preconditioner is not supported for models without material averaging. Disabling GMG preconditioner." << std::endl;
+      }
+    else if (dynamic_cast<const GeometryModel::EllipsoidalChunk<dim>*>(geometry_model.get()) != nullptr)
+      {
+        parameters.stokes_solver_type = Parameters<dim>::StokesSolverType::block_amg;
+
+        pcout << "Warning, the GMG preconditioner is not supported for models with ellipsoidal chunk geometry. Disabling GMG preconditioner." << std::endl;
+      }
+    else if (parameters.use_locally_conservative_discretization == true)
+      {
+        parameters.stokes_solver_type = Parameters<dim>::StokesSolverType::block_amg;
+
+        pcout << "Warning, the GMG preconditioner is not supported for models locally conservative discretization. Disabling GMG preconditioner." << std::endl;
+      }
+    else if (geometry_model->get_periodic_boundary_pairs().size() > 0)
```

Comment:
This is no longer true:v
https://github.com/geodynamics/aspect/blob/b2a9312199df918853086989fed6f4d2fb63b199/source/simulator/stokes_matrix_free.cc#L2389C1-L2410C6